pyro/infer/predictive.py

```diff
@@ -163,13 +180,15 @@ def get_samples(self, *args, **kwargs):
         """
         posterior_samples = self.posterior_samples
         if self.guide is not None:
-            # use return_sites='' as a special signal to return all sites
```

Comment:
I wonder if docs for `return_sites`: "by default only sample sites not present in `posterior_samples` are returned" is still valid now? The old behavior makes more sense to me.

Reply:
You mean for the case where we have both a guide and non-empty `posterior_samples`?

Reply:
I was thinking of actually raising an exception in that case, and adding support for it later if requested.

Reply:
I mean for the case `return_sites=None`, we will include all sites by default with this change. The former behavior is "only sample sites not present in posterior_samples are returned".